app/src/main/java/org/wikipedia/settings/Prefs.java

```diff
@@ -802,6 +802,22 @@ public static void setShouldShowSuggestedEditsTooltip(boolean enabled) {
         setBoolean(R.string.preference_key_show_suggested_edits_tooltip, enabled);
     }
 
+    public static boolean hasVisitedArticlePage() {
+        return getBoolean(R.string.preference_key_visited_article_page, false);
+    }
+
+    public static void setHasVisitedArticlePage(boolean visited) {
+        setBoolean(R.string.preference_key_visited_article_page, visited);
+    }
+
+    public static int fundraisingDialogShownInYear() {
+        return getInt(R.string.preference_key_fundraising_dialog_shown_in_year, 0);
+    }
+
+    public static void setFundraisingDialogShownInYear(int year) {
+        setInt(R.string.preference_key_fundraising_dialog_shown_in_year, year);
+    }
```

Comment:
This needs to be much more "general".  Calling it "fundraisingDialog" is too specific. Announcements could be made for all kinds of reasons, not just fundraising, and not just once per year.

If the goal is to remember which announcement(s) have already been shown, we need to store a list of announcement `id`s (taken from the `Announcement` object), and seeing if any of them match the requested announcement.

Reply:
Thanks! Done.